SecurityPkg/Library/Tpm2DeviceLibDTpm/Tpm2Svsm.c

```diff
@@ -0,0 +1,132 @@
+/** @file
+  SVSM TPM communication
+
+Copyright (C) 2024 James.Bottomley@HansenPartnership.com
+Copyright (C) 2024 IBM Corporation
+
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/Tpm2DeviceLib.h>
+#include <Library/PcdLib.h>
+#include <Library/AmdSvsmLib.h>
+
+#include "Tpm2DeviceLibDTpm.h"
+
+/**
+  Platform commands (MSSIM commands) can be sent through the
+  SVSM_VTPM_CMD operation. Each command can have its own
+  request and response structures.
+**/
+#define TPM_SEND_COMMAND  8
+
+/* Max req/resp buffer size */
+#define TPM_PLATFORM_MAX_BUFFER  4096
+
+STATIC UINT8  Tpm2SvsmBuffer[TPM_PLATFORM_MAX_BUFFER];
+
+#pragma pack(1)
+typedef struct _TPM2_SEND_CMD_REQ {
+  UINT32    Cmd;
+  UINT8     Locality;
+  UINT32    BufSize;
+  UINT8     Buf[];
+} TPM2_SEND_CMD_REQ;
+
+typedef struct _TPM2_SEND_CMD_RESP {
+  INT32    Size;

```

Comment:
I mean here we can use UINT32, see https://github.com/tianocore/edk2/pull/6527#discussion_r1882391835

Reply:
Changing `Size` to `UINT32` and removing the check for `<0` below.